appmap/test/test_configuration.py

```diff
@@ -19,3 +25,108 @@ def test_is_disabled_when_false(self, monkeypatch):
+        c = mocker.Mock()
+        c.__module__ = 'package1.package2'
+        c.__qualname__ = 'Mod1Class'
+        assert f.filter(c) is True
+
+    @pytest.mark.datafiles(
+        os.path.join(FIXTURE_DIR, 'appmap-func.yml')
+    )
+    def test_function_included(self, datafiles, mocker, monkeypatch):
+        monkeypatch.setenv("APPMAP", "true")
+        monkeypatch.setenv("APPMAP_CONFIG",
+                           os.path.join(str(datafiles), 'appmap-func.yml'))
+        monkeypatch.setenv("APPMAP_LOG_LEVEL", 'debug')
+
+        f = ConfigFilter(NullFilter())
+        c = mocker.Mock()
+        c.__module__ = 'package1.package2'
+        c.__qualname__ = 'Mod1Class.func'
+        assert f.filter(c) is True
+
+    @pytest.mark.datafiles(
+        os.path.join(FIXTURE_DIR, 'appmap.yml')
+    )
+    def test_function_included_by_package(self, datafiles, mocker, monkeypatch):
+        monkeypatch.setenv("APPMAP", "true")
+        monkeypatch.setenv("APPMAP_CONFIG",
+                           os.path.join(str(datafiles), 'appmap.yml'))
+        monkeypatch.setenv("APPMAP_LOG_LEVEL", 'debug')
+
+        f = ConfigFilter(NullFilter())
+        c = mocker.Mock()
+        c.__module__ = 'package1.package2'
+        c.__qualname__ = 'Mod1Class.func'
+        assert f.filter(c) is True
+
+    @pytest.mark.datafiles(
+        os.path.join(FIXTURE_DIR, 'appmap-class.yml')
+    )
+    def test_function_included_by_class(self, datafiles, mocker, monkeypatch):
+        monkeypatch.setenv("APPMAP", "true")
+        monkeypatch.setenv("APPMAP_CONFIG",
+                           os.path.join(str(datafiles), 'appmap-class.yml'))
+        monkeypatch.setenv("APPMAP_LOG_LEVEL", 'debug')
+
+        f = ConfigFilter(NullFilter())
+        c = mocker.Mock()
+        c.__module__ = 'package1.package2'
+        c.__qualname__ = 'Mod1Class.func'
+        assert f.filter(c) is True
+
+    @pytest.mark.datafiles(
+        os.path.join(FIXTURE_DIR, 'appmap.yml')
+    )
+    def test_class_prefix_doesnt_match(self, datafiles, mocker, monkeypatch):
+        monkeypatch.setenv("APPMAP", "true")
+        monkeypatch.setenv("APPMAP_CONFIG",
+                           os.path.join(str(datafiles), 'appmap.yml'))
+        monkeypatch.setenv("APPMAP_LOG_LEVEL", 'debug')
+
+        f = ConfigFilter(NullFilter())
+        c = mocker.Mock()
+        c.__module__ = 'package1_prefix'
+        c.__qualname__ = 'cls'
+        assert f.filter(c) is False
+
+    @pytest.mark.datafiles(
+        os.path.join(FIXTURE_DIR, 'appmap.yml')
+    )
+    def test_fn_prefix_doesnt_match(self, datafiles, mocker, monkeypatch):
+        monkeypatch.setenv("APPMAP", "true")
+        monkeypatch.setenv("APPMAP_CONFIG",
+                           os.path.join(str(datafiles), 'appmap.yml'))
+        monkeypatch.setenv("APPMAP_LOG_LEVEL", 'debug')
+
+        f = mocker.Mock()
+        f.__module__ = 'package1_prefix'
+        f.__qualname__ = 'cls.func'
+        fltr = ConfigFilter(NullFilter())
```

Comment:
Most of this is duplicated a lot which makes the code hard to follow. It seems most of it can be DRYed up.

Reply:
Yeah, I hadn't really come to grips with pytest fixtures yet when I wrote this. I understand them better now, and they seem like a good fit for cleaning this up.

Reply:
I added issue https://github.com/applandinc/appmap-python/issues/34 for this.